test/SizedArray.jl

```diff
@@ -68,6 +68,8 @@
         @test convert(Vector, SizedArray{Tuple{4}, Int, 1}(collect(3:6))) == collect(3:6)
         @test Matrix(SMatrix{2,2}((1,2,3,4))) == [1 3; 2 4]
         @test convert(Matrix, SMatrix{2,2}((1,2,3,4))) == [1 3; 2 4]
+        # Conversion after reshaping
+        @test_broken Array(SizedMatrix{2,2}([1,2,3,4])) == [1 3; 2 4]
```

Comment:
@andyferris I discovered this while writing `drop_sdims` and it seems quite problematic - any thoughts on how to fix it?  AFAICT we'll have to change the way that the SizedArray reshaping support works completely, and probably remove it from the normal constructors.

Actually the test here doesn't show the worst of it.  The bigger problem is that `SizedMatrix{2,3}` could wrap an Array of size `(3,2)`.